components/server/src/server.rs

```diff
@@ -209,9 +220,33 @@ pub fn run_tikv(
 
     dispatch_api_version!(config.storage.api_version(), {
         if !config.raft_engine.enable {
-            run_impl::<RocksEngine, API>(config, service_event_tx, service_event_rx)
+            if config.region_cache_memory_limit == ReadableSize(0) {
+                run_impl::<RocksEngine, RocksEngine, API>(
```

Comment:
I suggest always using HybridEngine and check the region cache limit innerly

Reply:
After discussed offline, we think this is riskless once it is not enabled.

Reply:
But using `HybridEngine` and when memory engine is not enabled, the code is actually a wrapper on RocksEngine without doing anything else

Reply:
It still needs to do some logics to check whether the memory engine can be used or something.